src/main/java/edu/tamu/app/controller/GetItForMeController.java

```diff
@@ -79,45 +84,94 @@ public ApiResponse getButtonsByBibId(@RequestParam(value="catalogName",defaultVa
      * @param String location Optional location parameter used for location-based redirect.
      * @return ApiResponse or RedirectView based on redirectUrl.
      */
     @RequestMapping("/get-buttons-redirect")
-    public Object getButtonsRedirectByBibId(@RequestParam(value="catalogName",defaultValue="evans") String catalogName, @RequestParam("bibId") String bibId, @RequestParam(value="verbose",defaultValue="false") boolean verbose, @RequestParam(value="location", required = false, defaultValue="") String location) {
-
-        Map<String,ButtonPresentation> buttonData = getItForMeService.getButtonDataByBibId(catalogName, bibId, verbose);
+    public Object getButtonsRedirectByBibId(@RequestParam(value="catalogName",defaultValue="evans") String catalogName, @RequestParam("bibId") String bibId, @RequestParam(value="location", required = false, defaultValue="") String location) {
+        bibId = ensureBibId(bibId);
+        Map<String,ButtonPresentation> buttonData = getItForMeService.getButtonDataByBibId(catalogName, bibId);
         String redirectUrl = null;
         if (buttonData != null) {
+            //test for the button set having a Cushing button
+            boolean hasCushing = false;
+            //test for the user requesting a Cushing button
+            boolean wantsCushing = location.equalsIgnoreCase("aeon");
+            Integer holdingsWithButtonsCount = 0;
+
             for (Map.Entry<String, ButtonPresentation> entry : buttonData.entrySet()) {
                 String holdingId = entry.getKey();
                 ButtonPresentation buttonPresentation = entry.getValue();
                 if (holdingId != null && buttonPresentation != null && buttonPresentation.getButtons() != null) {
+                    holdingsWithButtonsCount++;
                     for( Map<String, String> button: buttonPresentation.getButtons()) {
-                        String cssClass = button.get("cssClasses");
                         String linkHref = button.get("linkHref");
-                        boolean isCushing = ( cssClass != null && cssClass.contains("btn_cushing") ) &&
-                                            ( linkHref != null && linkHref.contains("aeon.library.tamu.edu") );
-
-                        if(isCushing && location.equalsIgnoreCase("aeon") && linkHref != null ) {
-                            redirectUrl = linkHref.startsWith("http") ? linkHref : "https://" + linkHref;
+                        //is the current button Cushing
+                        boolean isCushing = testIsCushing(button);
+                        //is at least one button in the set Cushing
+                        if (isCushing && !hasCushing) {
+                            hasCushing = true;
+                        }
+                        //when the user wants Cushing, grab the url of the first button that is for Cushing
+                        if(redirectUrl == null && isCushing && wantsCushing && linkHref != null ) {
```

Comment:
This is just a suggestion.

These if conditon blocks could probably be combined, kind of like this:
```java
                        // When the user wants Cushing, grab the url of the first button that is for Cushing.
                        // When the user doesn't want Cushing, grab the first button that isn't Cushing.
                        if (redirectUrl == null && linkHref != null && (!isCushing && !wantsCushing || isCushing && wantsCushing)) {
                            redirectUrl = buildRedirectUrl(linkHref);
                            break;
                        }
```